src/main/java/com/uniovi/services/TopicService.java

```diff
@@ -0,0 +1,26 @@
+package com.uniovi.services;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+
+import com.uniovi.entities.Incident;
+
+@Service
+public class TopicService 
+{
+	public static List<String> getTopicsOf(Incident incident)
+	{
+		List<String> aux = new ArrayList<String>();
+		if(!incident.getLocation().equals(null))
+			aux.add("geolocated");
+		//the withOperator topics is not implemented
+		if(!incident.getAgent().getKind().equals("sensor"))
+			aux.add("sensor");
+		if(aux.size()==0)
+			aux.add("standard");
```

Comment:
Here you can remove the `if(aux.size()==0)`. For now every incident will have the standard topic.

Reply:
Regarding the "withOperator" topic, operators can be stored in the properties map of the incident, so you can do something like this:
```java
for (String key : incident.getProperties().keySet()) {
    if ("operators".equals(key)) {
        aux.add("withOperator");
        break;
    }
}
```

Reply:
Also, it seems like you moved every test from 'src/test' to 'src/main/test', so please put the tests back to 'src/test' if you can.

Reply:
And finally, codacy seems to be complaining about `if(!incident.getLocation().equals(null))`, so use `==` instead. 
The Travis build is also failing but 'm pretty sure this is not your fault, so I will make some tests to see why it is failing. For now you can work on the previous comments while I try to fix Travis.